src/sentry/static/sentry/app/components/group/externalIssueActions.tsx

```diff
@@ -37,87 +37,146 @@ class ExternalIssueActions extends AsyncComponent<Props, State> {
     this.state = {
       showModal: false,
       action: 'create',
-      selectedIntegration: this.props.integration,
-      issue: this.getIssue(),
+      selectedIntegration: null,
       ...this.getDefaultState(),
+      ...this.linkedIssuesFilter(),
     };
   }
 
   getEndpoints() {
     return [];
   }
 
-  getIssue() {
-    return this.props.integration && this.props.integration.externalIssues
-      ? this.props.integration.externalIssues[0]
-      : null;
+  linkedIssuesFilter() {
+    return this.props.configurations.reduce(
+      (acc, curr) => {
+        if (curr.externalIssues.length) {
+          acc.linked.push(curr);
+        } else {
+          acc.unlinked.push(curr);
+        }
+        return acc;
+      },
+      {linked: [] as GroupIntegration[], unlinked: [] as GroupIntegration[]}
+    );
   }
 
-  deleteIssue(issueId: string) {
-    const {group, integration} = this.props;
-    const endpoint = `/groups/${group.id}/integrations/${integration.id}/?externalIssue=${issueId}`;
+  deleteIssue(integration: GroupIntegration) {
+    const {group} = this.props;
+    const {externalIssues} = integration;
+    let issue = externalIssues[0];
+    let {id} = issue;
+    const endpoint = `/groups/${group.id}/integrations/${integration.id}/?externalIssue=${id}`;
+
     this.api.request(endpoint, {
       method: 'DELETE',
       success: () => {
         addSuccessMessage(t('Successfully unlinked issue.'));
+        let unlinked = JSON.parse(JSON.stringify(integration)) as GroupIntegration;
```

Comment:
@NisanthanNanthakumar What's up with calling `JSON.parse` and `JSON.stringify` like this?

Reply:
@scefali ahh this is Javascript deep copy hack. I just checked and we are already using lodash. So I'll refactor to use lodash.deepCopy